x-pack/legacy/plugins/uptime/public/state/api/monitor.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ThrowReporter } from 'io-ts/lib/ThrowReporter';
+import { getApiPath } from '../../lib/helper';
+import { MonitorDetailsType, MonitorDetails } from '../../../common/runtime_types';
+
+interface ApiRequest {
+  monitorId: string;
+  basePath: string;
+}
+
+export const fetchMonitorDetails = async ({
+  monitorId,
+  basePath,
+}: ApiRequest): Promise<MonitorDetails> => {
+  const url = getApiPath(`/api/uptime/monitor/details?monitorId=${monitorId}`, basePath);
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(response.statusText);
+  }
+  return response.json().then(data => {
+    ThrowReporter.report(MonitorDetailsType.decode(data));
```

Comment:
This is a question because I'm still learning `io-ts`. If our API response data fails to `decode` correctly, `ThrowReporter` will dispatch an error that breaks the control flow of this function?

In my own code, I'm doing something like:
```typescript
const decoded = MyType.decode(data);
ThrowReporter.report(decoded);
if (isRight(decoded)) {
  return decoded.right;
}
throw new Error('some message');
```

It seems like the example in my comment is a more verbose way of doing the same thing if `ThrowReporter` will throw us out of the current function.